### PR TITLE
[LOOP-28] Upload loop settings data to Tidepool backend

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-28
- Add device to CGMManagerStatus
- Make CGMManagerStatus Equatable and Codable
- Maintain latest settings in SettingsStore
- Add notification settings, and controller, pump, and cgm devices to StoredSettings
- Add delay to StoredInsulinModel
- Remove device token from StoredSettings
- Refactor StoredSettings.syncIdentifier to use UUID type
- Add tests

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR for build: https://github.com/tidepool-org/LoopWorkspace/pull/698

File: LoopKit/DeviceManager/CGMManager.swift [hunk 1]
```diff
@@ -20,15 +20,40 @@ public enum CGMReadingResult {
...
 }
 
-public struct CGMManagerStatus {
+public struct CGMManagerStatus: Equatable {
     // Return false if no sensor active, or in a state where no future data is expected without user intervention
     public var hasValidSensorSession: Bool
 
     public var lastCommunicationDate: Date?
+
+    public var device: HKDevice?
```
Comment (ps2) [inline LoopKit/DeviceManager/CGMManager.swift:29]: How does this differ from the `device` attribute on the CGMManager itself?  What would it mean if this returned a different device than the one on cgmmanager? Which one should the hosting app use?

**Review by ps2 — COMMENTED** [review]
Could you explain the new HKDevice attribute on CGMManagerStatus? It raises some questions for me.

File: LoopKit/DeviceManager/CGMManager.swift (hunk 1 above)
Comment (darinkrauss) [reply LoopKit/DeviceManager/CGMManager.swift:29]: It is the same. I don't know why, but the `PumpManager` returns the device as part of the `PumpManagerStatus`, but the `CGMManager` returns the device on its own. This just makes it uniform (so all "status" from the manager goes through the `*Status` object). I'm happy to remove the device on the `CGMManager` itself, but I wanted to make as few changes as possible.

Comment (ps2) [reply LoopKit/DeviceManager/CGMManager.swift:29]: Ok, I don't think we should have two different ways of doing this on CGMManager. As for which one is better, I don't have a strong opinion. Having it on status means that changes to device can be 'events', so clients don't have to poll for changes, if they are concerned with that.  But it also means additional costs in terms of including infrequently changed data on a frequently passed struct.

**Review by rickpasetto — COMMENTED** [review]
LGTM, but defer to @ps2 on approval.

File: LoopKit/DeviceManager/CGMManager.swift (hunk 1 above)
Comment (darinkrauss) [reply LoopKit/DeviceManager/CGMManager.swift:29]: Updated. Removed `device` on `CGMManager` protocol. Also updated `SupportInfoProvider` protocol to use `CGMManagerStatus`.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**